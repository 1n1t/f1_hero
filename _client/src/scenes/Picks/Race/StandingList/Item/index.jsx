import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const Item = ({ score, position, _driver }) => {
	const icon =
		score !== null ? (
			<div className="score">{score}pt</div>
		) : (
			<i className="fa fa-sort" aria-hidden="true" />
		);

	return (
		<div className="StandingItem">
			<div className="position">{position}.</div>
			<div className="driver-name">{_driver.name}</div>
			<div className="driver-code">{_driver.code}</div>
			{icon}
		</div>
	);
};

Item.defaultProps = {
	score: null
};

Item.propTypes = {
	score: PropTypes.number,
	position: PropTypes.number.isRequired,
	_driver: PropTypes.shape({
		name: PropTypes.string.isRequired,
		code: PropTypes.string.isRequired
	}).isRequired
};

export default Item;
