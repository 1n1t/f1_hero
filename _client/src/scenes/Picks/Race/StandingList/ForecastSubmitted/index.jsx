import React from 'react';
import PropTypes from 'prop-types';

import Item from '../Item';
import { listItemPropTypes } from '../index';

const ForecastSubmitted = ({ items }) => (
	<div className="standings-container">
		{items.map(item => <Item key={item.position} {...item} />)}
	</div>
);

ForecastSubmitted.propTypes = {
	items: PropTypes.arrayOf(PropTypes.shape(listItemPropTypes)).isRequired
};

export default ForecastSubmitted;
