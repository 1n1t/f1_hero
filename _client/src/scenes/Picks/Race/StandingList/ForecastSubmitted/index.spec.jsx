import r from 'Root/jest/utils/renderer';
import pickSample from 'JestClient/samples/pick';

import ForecastSubmitted from './index';
import Item from '../Item';

const defaultProps = {
	items: pickSample.forecast
};

r.init(ForecastSubmitted, defaultProps);

it('renders correctly', () => {
	const wrapper = r.render(null, true);
	expect(wrapper).toMatchSnapshot();
});

it('renders correct amount of list items', () => {
	const wrapper = r.render(null, true);
	const items = wrapper.find(Item);

	expect(items).toHaveLength(defaultProps.items.length);
});
