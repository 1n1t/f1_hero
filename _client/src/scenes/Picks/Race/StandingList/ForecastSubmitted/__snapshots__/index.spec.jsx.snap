// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="standings-container"
>
  <Item
    _driver={
      Object {
        "code": "HAM",
        "name": "L. Hamilton",
      }
    }
    key="1"
    position={1}
    score={18}
  />
  <Item
    _driver={
      Object {
        "code": "RIC",
        "name": "D. Ricciardo",
      }
    }
    key="2"
    position={2}
    score={5}
  />
  <Item
    _driver={
      Object {
        "code": "VAN",
        "name": "S. Vandoorne",
      }
    }
    key="3"
    position={3}
    score={0}
  />
</div>
`;
